libtenzir/src/tql2/eval_binary.cpp

```diff
@@ -541,6 +541,30 @@ struct EvalBinOp<Op, string_type, string_type> {
   }
 };
 
+template <ast::binary_op Op>
+  requires(Op == ast::binary_op::eq || Op == ast::binary_op::neq)
+struct EvalBinOp<Op, record_type, record_type> {
+  static auto eval(const arrow::StructArray& l, const arrow::StructArray& r,
+                   auto&&) -> std::shared_ptr<arrow::BooleanArray> {
+    constexpr auto invert = Op == ast::binary_op::neq;
+    auto b = arrow::BooleanBuilder{};
+    check(b.Append(l.Equals(r) != invert));
```

Comment:
This will add a single boolean for every batch of events, instead of adding one boolean per event in each batch. This works if the batch size is 1 by accident, and will probably crash in all other cases.